packages/web/src/pages/sign-up-page/utils/useDetermineAllowedRoutes.ts

```diff
@@ -81,6 +81,9 @@ export const useDetermineAllowedRoute = () => {
     // If requested route is allowed return that, otherwise return the last step in the route stack
     const correctedPath = isAllowedRoute
       ? attemptedPath
+      : // IF we attempted to go to /signup directly, that means it was a link from somewhere else in the app, so we should start back at the beginning
+      attemptedPath === '/signup'
+      ? allowedRoutes[0]
```

Comment:
I dont "love" this as the fix but the reasoning is that right now all the paths from elsewhere in the app just go to `/signup` (and also the link from sign-in for some reason) and we just let the rerouting logic take over.
By default rerouting always goes to the furthest page you were at based on your redux stuff.
So basically this instead makes the route logic look like:
* Tried to route to `/sign-up` directly => route to email page always (form values are preserved if they go back thru stuff tho)
* Tried to route to a specific page than where I'm at (valid route specifically) => route to furthest page
* Tried to route to a junk route that doesnt exist (i.e. `/signup/junk`) => blank white page 😓  (want to do something about this but its not the most pressing rn I suppose)

Reply:
I did consider making routes[0] always the default but I think that would require reworking our logic? I think rn we kind of use that "go to latest" in a spot or two to get users routed to the right spot lol

Reply:
ah yeah we would have to rework the "loading screen" 🤔, maybe its better to that than this hack tho